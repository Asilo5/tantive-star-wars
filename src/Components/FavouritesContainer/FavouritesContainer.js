import React from 'react';
import { Link } from 'react-router-dom';
import './FavouritesContainer.scss';
// import CharacterCard from '../CharacterCard/CharacterCard';
import { FaJedi } from "react-icons/fa";


const FavouritesContainer = ({ id, name, homeworld, population, species,toggleFavourites, favourites }) => {
    if(favourites.length === 0) {
      return (
        <>
        <h2 className='fav_h2'>Want to see your favorite character!</h2>
          <h3 className='fav_h3'>Go back to movies and select a few first.</h3>
          <Link to='/movies'>
            <button className='back_to_movies__btn'>Back to Movies</button>
          </Link>
        </>
     ) 
    } 
    const theFilm = favourites.map((favourite, i) => <p key={i}>{favourite}</p>);
    const toggleIcon = favourites.includes(id) ? 'favourite-icon' : 'jedi-icon';
    return (
      <section className='card'>
        <p className={toggleIcon} onClick={() => toggleFavourites(id)}><FaJedi/></p>
        <h2>{name}</h2>
        <h3>Homeworld:</h3>
        <p>{homeworld}</p>
        <h3>Population:</h3>
        <p>{population}</p>
        <h3>Species: </h3>
        <p>{species}</p>
        <h3>Seen in:</h3>
        {theFilm}
      </section>
  ) 
}



export default FavouritesContainer;