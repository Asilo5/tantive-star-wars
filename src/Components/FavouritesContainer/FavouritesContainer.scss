$primary-color: green;
$hover-color: gold;

.blank_error {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    padding: 20px;
}

.favourite {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    overflow: scroll;
    height: 56em;
    padding: 20px;
}

.fav_h2 {
    color: white;
    font-size: 3em;
}

.fav_h3 {
    color: white;
    font-size: 2em;
}

.back_to_movies__btn {
    border-radius: 10px;
    height: 55px;
    font-size: 2em;
    margin: 5px;
    padding: 13px;
    border: 2px solid transparent;
    cursor: pointer;
    &:hover,
      &:focus {
         border: 2px solid $primary-color;
         color: $primary-color;
      }
}
