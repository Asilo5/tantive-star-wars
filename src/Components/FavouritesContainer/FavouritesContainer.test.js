import React from 'react';
import { shallow } from 'enzyme';
import FavouritesContainer from './FavouritesContainer';

describe('FavouritesContainer', () => {
  let wrapper
  const mockChangeSelectedMovie = jest.fn();
  const mockSetFavorite = jest.fn();

  it('should match the snapshot with all the data passed in correctly', () => {
      wrapper = shallow(<FavouritesContainer
          characters={[]}
          changeSelectedMovie={mockChangeSelectedMovie}
          isReady={false}
          setFavorite={mockSetFavorite}
          favoriteList={[{}, {}]} />)
      expect(wrapper).toMatchSnapshot();
  });

});