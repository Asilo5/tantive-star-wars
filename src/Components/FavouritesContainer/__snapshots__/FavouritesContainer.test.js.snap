// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesContainer should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="favourite"
/>
`;
